wordpress-mobile/WordPress-Android
# Jetpack Focus: Add jetpack feature card skeleton

Parent #17336 

This PR implements the skeleton pieces for showing the JetpackFeatureCard. 


Light | Dark
-- | --
<img width="250" height="500" alt="Light" src="https://user-images.githubusercontent.com/506707/208523786-15cdcb9e-0949-4ab9-8717-3760ee509265.png"> | <img width="250" height="500" alt="Dark" src="https://user-images.githubusercontent.com/506707/208523799-8c0f765f-8b01-4eb1-8d1f-56dc524f0e8f.png">

Light Menu | Dark Menu
-- | --
<img width="250" height="500" alt="Light" src="https://user-images.githubusercontent.com/506707/208523948-b31c0407-9cc4-4892-bf2f-0ab15cdc934b.png"> | <img width="250" height="500" alt="Dark" src="https://user-images.githubusercontent.com/506707/208523957-bd27810e-ec78-4c22-86f2-7e41720c5328.png">


**Includes**
UI Components: view and menu layouts, icons and strings
Tracking Events + card shown tracker
New card type `JetpackFeatureCard`
New view holder and updates to the MySiteAdapter and callback adapter
Updates to MySiteViewModel to actually show the card and track the events

**Not Included**
CTA's for the card

Note: `MySiteViewModel` suppresses ForbiddenComments because I left todo's for the next unit of work.

**To test:**
Test 1: Card is not shown within Jetpack App
- Install JP app
- Login 
- Navigate to Me -> App Settings -> Debug Settings
- If `jp_removal_three` is enabled, please disable and restart the app
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is not shown

Test 2: Card is not shown with WordPress App when phase 3 flag is off
- Install WP app
- Login 
- Navigate to Me -> App Settings -> Debug Settings
- If `jp_removal_three` is enabled, please disable and restart the app
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is shown

Test 3: Card is shown when WordPress App and phase 3 flag is on
- Install WP app
- Login 
- Navigate to Me -> App Settings -> Privacy Settings and turn on Collection Information
- Navigate to -> Debug Settings -> 
- Enable `jp_removal_three` and restart the app
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is shown

Test 4: Card clicks are tracked
- Run test 3
- ✅ Verify the log contains 🔵 Tracked: remove_feature_card_displayed, Properties: {"phase":"three"}
- Tap on the Jetpack Feature Card
- ✅ Verify the log contains 🔵 Tracked: remove_feature_card_tapped, Properties: {"phase":"three"}
- Tap on "Learn More" link
- ✅ Verify the log contains 🔵 Tracked: remove_feature_card_link_tapped, Properties: {"phase":"three"}
- Tap on the more menu
- ✅ Verify the log contains 🔵 Tracked: remove_feature_card_menu_accessed, Properties: {"phase":"three"}
- Tap on the more menu -> remind me later item
- ✅ Verify the log contains 🔵 Tracked: remove_feature_card_remind_later_tapped, Properties: {"phase":"three"}
- Tap on the more menu -> hide item
- ✅ Verify the log contains  🔵 Tracked: remove_feature_card_hide_tapped, Properties: {"phase":"three"}


## Regression Notes
1. Potential unintended areas of impact
The Jetpack feature card shows when it should not

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated `MySiteViewModelTest`

PR submission checklist:
